File: app/src/main/java/com/recen/learn/practice1/Practice1.java
package com.recen.learn.practice1;

import android.arch.lifecycle.MutableLiveData;
import android.databinding.ObservableField;

import com.recen.dotframe.impl.PageWrapper;

public class Practice1 extends PageWrapper<Practice1Repository> {
    public ObservableField<String> formatedPhone = new ObservableField<>();
    public ObservableField<String> smsCode = new ObservableField<>();
    @Override
    protected void initData() {
        //进入页面后的数据初始化

    }

    public void smsCodeBtnClick(){
        page.showMessage("点击了获取验证码");
        mRepository.getData();
    }

    public void voiceSmsCodeBtnClick(){
        page.showMessage("点击了获取语音验证码");
    }

    public void submitBtnClick(){
        page.showMessage("点击了登录");
    }
}

File: app/src/main/java/com/recen/learn/practice1/Practice1.kt
package com.recen.learn.practice1

import android.databinding.ObservableArrayList
import android.databinding.ObservableField

import com.recen.dotframe.impl.PageWrapper
import com.recen.dotframe.interfaces.RepoCallback
import com.recen.learn.download.update.Update
import com.recen.learn.model.ReadData

import java.util.ArrayList

class Practice1 : PageWrapper<Practice1Repository>() {

    var dataList = ObservableArrayList<PracticeItem>()
    var textField = ObservableField("--")
    override fun initData() {
        mRepository.getData(object : RepoCallback<List<ReadData>> {
            override fun onSuccess(data: List<ReadData>) {
                val list = ArrayList<PracticeItem>()
                for (item in data) {
                    list.add(PracticeItem(item, page))
                }
                dataList.addAll(list)
            }

            override fun onFailure() {

            }
        })
    }

    fun sendSmsCode() {

    }

    fun login() {
        page.showMessage("login")
    }

}
